marlin-succinct/circuits/src/compiled.rs

```diff
@@ -0,0 +1,117 @@
+/*****************************************************************************************************************
+
+This source file implements the compiled constraints primitive.
+
+*****************************************************************************************************************/
+
+use sprs::CsMat;
+use commitment::urs::URS;
+use algebra::{Field, PrimeField, PairingEngine};
+use ff_fft::{DensePolynomial, Evaluations, EvaluationDomain};
+pub use super::index::Index;
+
+pub struct Compiled<E: PairingEngine>
```

Comment:
We will need the row/col/val polynomials evaluated on multiple domains. Check out the `Index` type [here](https://github.com/scipr-lab/marlin/blob/master/src/ahp/indexer.rs). I think it should be factored better into 3 records, one for each of A, B, C,  but I do believe we need all the values included in that type.

Reply:
OK, investigating this.